cloudmanager/automation_config.go

```diff
@@ -0,0 +1,212 @@
+// Copyright 2019 MongoDB Inc
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cloudmanager
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+
+	atlas "github.com/mongodb/go-client-mongodb-atlas/mongodbatlas"
+)
+
+const (
+	automationBasePath = "groups/%s/automationConfig"
+)
+
+// AutomationService is an interface for interfacing with the Automation Config
+// endpoints of the MongoDB CLoud API.
+// See more: https://docs.cloudmanager.mongodb.com/reference/api/automation-config/
+type AutomationService interface {
+	Get(context.Context, string) (*AutomationConfig, *atlas.Response, error)
+	Update(context.Context, string, *AutomationConfig) (*AutomationConfig, *atlas.Response, error)
+}
+
+// AutomationServiceOp handles communication with the Automation config related methods of the MongoDB Cloud API
+type AutomationServiceOp struct {
+	client *Client
+}
+
+// See more: https://docs.cloudmanager.mongodb.com/reference/api/automation-config/#get-the-automation-configuration
+func (s *AutomationServiceOp) Get(ctx context.Context, groupID string) (*AutomationConfig, *atlas.Response, error) {
+	basePath := fmt.Sprintf(automationBasePath, groupID)
+
+	req, err := s.client.NewRequest(ctx, http.MethodGet, basePath, nil)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	root := new(AutomationConfig)
+	resp, err := s.client.Do(ctx, req, root)
+	if err != nil {
+		return nil, resp, err
+	}
+
+	return root, resp, err
+}
+
+// See more: https://docs.cloudmanager.mongodb.com/reference/api/automation-config/#update-the-automation-configuration
+func (s *AutomationServiceOp) Update(ctx context.Context, groupID string, updateRequest *AutomationConfig) (*AutomationConfig, *atlas.Response, error) {
+	basePath := fmt.Sprintf(automationBasePath, groupID)
+
+	req, err := s.client.NewRequest(ctx, http.MethodPatch, basePath, updateRequest)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	root := new(AutomationConfig)
+	resp, err := s.client.Do(ctx, req, root)
+	if err != nil {
+		return nil, resp, err
+	}
+
+	return root, resp, err
+}
+
+var _ AutomationService = new(AutomationServiceOp)
+
+type AutomationConfig struct {
+	AgentVersion       *map[string]interface{}   `json:"agentVersion,omitempty"`
```

Comment:
Right now I don't have much use for many of the structs here so I'm not going to complicate this PR trying to type everything, right now we only care about `Processes` and `ReplicaSets`